docs/lifecycle-management.md

```diff
@@ -0,0 +1,185 @@
+# Higher level VM Management
+
+## Motivation
+
+So far KubeVirt provides a pod-like interface when managing VMs. This seems to
+be a good building block.
+On the other hand this might not be sufficient for usual virtualization
+management UIs, which do expect to manage the complete life-cycle of a VM,
+including shut down VMs 8which is currently not intended by the core runtime).
+
+
+This document outlines how minimal assumptions can be met and also visits how
+other requirements impact the design.
+
+
+## Requirements
+* Running and non-running VMs are handled through the same API
+* Creating a VM must be separate from launching a VM
+* Support for “flavor” based VMs (templating)
```

Comment:
Maybe before discussing the value of each it makes sense to differentiate the words:

* templating
* instance type - @mykaul how do you see the difference to a template?
* flavor - @berrange could you define flavor in the ctx of openstack?

I'm using the word templating in the sense of a pre-defined vm configuration, which can be used to create an instance.